NitroxClient/GameLogic/Vehicles.cs

```diff
@@ -428,5 +429,16 @@ public bool RemoveVehicle(NitroxId id)
         {            
             return (T)vehiclesById[vehicleId];
         }
+
+        public Optional<T> TryGetVehicle<T>(NitroxId vehicleId) where T : VehicleModel
+        {
+            VehicleModel vehicle;
+            vehiclesById.TryGetValue(vehicleId, out vehicle);
+            if (vehicle == null)
+            {
+                return Optional.Empty;
+            }
+            return Optional.Of((T)vehicle);
```

Comment:
You can do OfNullable

Reply:
I wanted to get the cast exception in case it's not null but wrong/unexpected type so I have to do checks seperately.